nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp

```diff
@@ -450,7 +456,13 @@ bool RegulatedPurePursuitController::inCollision(const double & x, const double
 double RegulatedPurePursuitController::costAtPose(const double & x, const double & y)
 {
   unsigned int mx, my;
-  costmap_->worldToMap(x, y, mx, my);
+  auto convertWorldToMap = costmap_->worldToMap(x, y, mx, my);
+  if (!convertWorldToMap) {
```

Comment:
This function is only used when the x,y represents our robot pose, this should never be out of bounds.

Reply:
So I think these changes in this function can be reverted?

Reply:
I think it is not a good idea. If you remember the second issue that I brought up in #2332 , these changes actually solves that issue. 

>  This function is only used when the x,y represents our robot pose, this should never be out of bounds.

Explaining it further: With the parameters set in that issue, the current position of the robot can be sometimes outside of the local costmap. Then again, we have a seg fault. In order to avoid it, we check for the bounds.

Reply:
Please give me an example flow of data that this would prevent?  What I see is:
- `costAtPose` is only used as an argument for `applyConstraints()`
- the `pose` given to `applyConstraints` is `pose`, which in that function I track to the argument `pose` https://github.com/ros-planning/navigation2/blob/8595b870dd66ef709101786a13eb86a5fc889e76/nav2_regulated_pure_pursuit_controller/src/regulated_pure_pursuit_controller.cpp#L243
- that pose represents the robot's current pose -- so there's no look ahead applied and my assumption is that at all times the robot should be on the costmap

Reply:
That seems.... bad. I know you set it to 1x1 to prove a point, but what if this was a more normal size like 5x5 or a more reasonable update rate like 5hz? We should cover the crash, you bring up a good point, but those are some extraordinarily poor configurations :laughing:. I think we just need to change the logging error message